firstvoices/backend/views/sites_views.py

```diff
@@ -72,27 +93,53 @@ def list(self, request, *args, **kwargs):
         Override the list method to group sites by language.
         """
         # custom queryset to avoid prefetching from unneeded tables (list view has less detail)
+        # note that the titles are converted to uppercase and then sorted which will put custom characters at the end
         queryset = Site.objects.select_related("language").order_by(
             Upper("language__title"), Upper("title")
         )
 
         # apply permissions
         sites = utils.filter_by_viewable(request.user, queryset)
 
-        # serialize each site (groupby can't handle Models)
-        site_jsons = [
-            SiteSummarySerializer(site, context={"request": request}).data
-            for site in sites
-        ]
-
-        # group by language
-        rows = groupby(site_jsons, itemgetter("language"))
-        data = [
-            {
-                "language": language if language is not None else "Other",
-                "sites": list(items),
-            }
-            for language, items in rows
-        ]
+        data = group_sites_by_language(request, sites)
 
         return Response(data)
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="A list of language sites a given user is a member of. If "
+        "there are no accessible sites or the user does not have "
+        "membership to any site then the list will be empty.",
+        responses={
+            200: inline_serializer(
+                name="InlineLanguageSerializer",
+                fields={
+                    "language": serializers.CharField(),
+                    "sites": SiteSummarySerializer(many=True),
+                },
+            ),
+        },
+    ),
+)
+class MySitesViewSet(
+    AutoPermissionViewSetMixin, mixins.ListModelMixin, viewsets.GenericViewSet
+):
+    """
+    Summary information about language sites a given user has membership to.
+    """
+
+    http_method_names = ["get"]
+    pagination_class = None
+    serializer_class = SiteSummarySerializer
+
+    def get_queryset(self):
+        # get the site objects filtered by the membership set for the user
+        # note that the titles are converted to uppercase and then sorted which will put custom characters at the end
+        queryset = Site.objects.filter(membership_set__user=self.request.user).order_by(
+            Upper("title")
+        )
```

Comment:
Probably still want to select_related(language) since it is used in the serializer

Reply:
Good point. I have added it back in. Thanks.